src/shared/watchdog.c

```diff
@@ -324,16 +324,25 @@ static int open_watchdog(void) {
          * has the benefit that we can easily find the matching directory in sysfs from it, as the relevant
          * sysfs attributes can only be found via /sys/dev/char/<major>:<minor> if the new-style device
          * major/minor is used, not the old-style. */
-        fn = !watchdog_device || path_equal(watchdog_device, "/dev/watchdog") ?
-                "/dev/watchdog0" : watchdog_device;
+        try_order = !watchdog_device || path_equal(watchdog_device, "/dev/watchdog") ?
```

Comment:
why not `PATH_IN_SET(watchdog, "/dev/watchdog", "/dev/watchdog0")` here?

i.e. I'd always consider these devices equivalent, and given they are equivalent we want to use the better one, always, but if the better one doesn't exist the worse one.

Reply:
Simply I forgot to apply that from your suggestion :-)